plugins/admin/pkg/command/registry/remove.go

```diff
@@ -0,0 +1,148 @@
+// limitations under the License.
+
+package registry
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/client-contrib/plugins/admin/pkg"
+
+	"github.com/spf13/cobra"
+)
+
+var username string
+var server string
+
+// NewPrRmCommand represents the remove command
+func NewPrRmCommand(p *pkg.AdminParams) *cobra.Command {
+	var prRmCmd = &cobra.Command{
+		Use:   "remove",
+		Short: "Remove registry with credentials",
+		Long: `Remove registry with credentials that used for Service deployment
+For example:
+
+kn admin registry remove \
+  --username=[REGISTRY_USER] \
+  --server=[REGISTRY_SERVER_URL]`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if username == "" {
+				return errors.New("'registry remove' requires the registry username provided with the --username option")
+			}
+			if server == "" {
+				return errors.New("'registry remove' requires the registry server url provided with the --server option")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// get all credential secrets which have the label managed-by=kn-admin-registry
+			secrets, err := p.ClientSet.CoreV1().Secrets("default").List(metav1.ListOptions{
+				LabelSelector: labels.SelectorFromSet(AdminRegistryLabels).String(),
+			})
+			if err != nil {
+				return fmt.Errorf("failed to list secret: %v", err)
+			}
+
+			// filter the secrets with username and server
+			secretsMap := make(map[string]*corev1.Secret)
+			for _, secret := range secrets.Items {
+				registry := Registry{}
+				err = json.Unmarshal(secret.Data[DockerJSONName], &registry)
+				if err != nil {
+					return fmt.Errorf("failed unmarshal secret data '.dockerconfigjson': %v", err)
+				}
+				for secretServer, secretAuth := range registry.Auths {
+					if secretServer == server && secretAuth.Username == username {
+						secretsMap[secret.Name] = &secret
+					}
+				}
+			}
+			if len(secretsMap) == 0 {
+				cmd.Printf("No registry found for server: '%s' and username: '%s'\n", server, username)
+				return nil
+			}
+
+			defaultSa, err := p.ClientSet.CoreV1().ServiceAccounts("default").Get("default", metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to get serviceaccount: %v", err)
+			}
+
+			desiredSa := defaultSa.DeepCopy()
+			imagePullSecrets := []corev1.LocalObjectReference{}
+			for _, ips := range desiredSa.ImagePullSecrets {
+				if _, ok := secretsMap[ips.Name]; !ok {
+					// only store the secrets that do not exist in the map
+					imagePullSecrets = append(imagePullSecrets, ips)
+				}
+
+			}
+
+			desiredSa.ImagePullSecrets = imagePullSecrets
+			_, err = p.ClientSet.CoreV1().ServiceAccounts("default").Update(desiredSa)
+			if err != nil {
+				return fmt.Errorf("failed to remove registry secret in default serviceaccount: %v", err)
```

Comment:
serviceaccount => ServiceAccount

Reply:
Fixed along with all the other parts that using "serviceaccount" also.